test/useTranslation.loading.spec.js

```diff
@@ -49,73 +15,32 @@ describe('useTranslation loading ns', () => {
     newI18n.use(backend).init({
       lng: 'en',
       fallbackLng: 'en',
-
       interpolation: {
         escapeValue: false, // not needed for react!!
       },
     });
   });
 
-  const TestElement = ({ useSuspense = true }) => {
-    const [t] = useTranslation('common', { i18n: newI18n, useSuspense });
-    return <div>{t('key1')}</div>;
-  };
-
-  it('should wait for correct translation', done => {
-    const spy = sinon.spy();
-
-    console.error = jest.fn(); // silent down the error boundary error from react-dom
-    let wrapper = mount(
-      <ErrorBoundary spy={spy}>
-        <TestElement />
-      </ErrorBoundary>,
+  it('should wait for correct translation with suspense', async () => {
+    const { result, waitForNextUpdate } = renderHook(() =>
+      useTranslation('common', { i18n: newI18n, useSuspense: true }),
     );
-
-    expect(wrapper.contains(<div>Something went wrong.</div>)).toBe(true);
-    expect(spy.callCount).toBe(1);
-
+    expect(result.current).toBe(null);
```

Comment:
```suggestion
    expect(result.all).toHaveLength(0);
```

react `suspense` is a very strange feature. To make suspense works a throw exception is needed.

```js
 throw new Promise(resolve => {
    loadNamespaces(i18n, namespaces, () => {
      resolve();
    });
  });
```

The throw exception will "pause` the render displaying a fallback component until the promise is finished. 

So, testing with `@testing-library/react-hooks` means that on the first render the result doesn't contain any ref to the current rendered DOM because a fallback component is displayed. You can check that no renders has been done because `result.all` array is empty. 

`result.current` as `result.error` will try to return the first element of the array but it doesn't exist yet.

Reply:
I will commit the suggestion if no one has any objections?